src/Core/Graphics/Sprite.cpp

```diff
@@ -525,6 +549,7 @@ namespace obe::Graphics
         this->setPositionTransformer(positionTransformer);
         this->setLayer(layer);
         this->setZDepth(zdepth);
+        this->setColor(spriteColor);
```

Comment:
Did you test the case where you put no color ?
Since you use the default constructor of the Color object, the default color will be black, which means every pixel of the Sprite will be black.
You should either change the `spriteColor` construction to
```cpp
Color spriteColor = Color::White;
```
or apply the `this->setColor` only if a color attribute is present.